app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -45,23 +48,34 @@ def supports_authentication?(authtype)
     supported_auth_types.include?(authtype.to_s)
   end
 
-  def self.raw_connect(server, port, username, password, service = "Service")
-    require 'ovirt'
+  def raw_connect(server, port, username, password, service = "Service")
```

Comment:
This should stay as a class method that knows nothing about the database record.  Its purpose is for doing raw connections when you don't have a record (mostly for diagnostics).  Instead, add a `path` parameter to the signature here, and change the `#connect` method to do the logic with the instance's column values.